### PR TITLE
Add "No results found" utility to search dropdown

### PR DESCRIPTION
**Pull request by Animesh239**
close: #676 

## Description

added: no results found for "value" in the search-bars

## Screenshots : (for networkSearchBar)
Screenshot 2023-12-20 151901
Screenshot 2023-12-20 151929

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by Animesh239** [comment]
there is a bug in my PR , will fix it soon...

**Comment by Animesh239** [comment]
@dpgiakatos you can review this PR

**Review by dpgiakatos — CHANGES_REQUESTED** [review]
Hello,
This pull request aims to resolve the issue when we encounter no results in a search bar. It includes modifications in two components:
1. **NetworkSearchBar.vue**: This component manages the search functionality for networks or countries. The changes made have been tested successfully and work without any issues. However, it's worth noting that this component may be deprecated and replaced with another one after the completion of the IYP integration, which is currently in progress.
2. **LocationSerachBar.vue**: This component handles the search for an AS network. The changes made have also been tested, but they cause a crash in the application. I've attached a screenshot with the test case and the logs from the browser console.
image
[#708-LocationSearchBar.log](https://github.com/InternetHealthReport/ihr-website/files/13751635/708-LocationSearchBar.log)

In summary, this pull request cannot be merged without first addressing the issues mentioned in this review.

**Comment by dpgiakatos** [comment]
When you have addressed the issue, please request a review.

**Comment by Animesh239** [comment]
hey @dpgiakatos 
I made the necessary changes to the locationSearchBar file in the prev commit.
Please review my PR and let me know if any additional changes are needed.

File: src/components/search/LocationSearchBar.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <script setup>
 import { QSelect, QIcon, QSpinner, QItem, QItemSection } from 'quasar'
 import { NetworkDelayLocation, NetworkQuery } from '@/plugins/IhrApi'
-import { ref, inject, watch } from 'vue'
+import { ref, inject, watch, defineProps, defineEmits } from 'vue'
```
Comment (dpgiakatos) [inline src/components/search/LocationSearchBar.vue:4]: defineProps and defineProps are a compiler macro and no longer needs to be imported.

File: src/components/search/LocationSearchBar.vue [hunk 2]
```diff
@@ -27,21 +27,11 @@ const props = defineProps({
...
-  'select': function(location) {
-    if (location !== null) {
-      return true;
-    } else {
-      console.warn('Location is missing!');
-      return false;
-    }
-  }
-})
+const emits = defineEmits(['select'])
```
Comment (dpgiakatos) [inline src/components/search/LocationSearchBar.vue:30]: Since this component is the last one that uses the select emit function, this function should be initiated here. Please undo this change.

**Review by dpgiakatos — CHANGES_REQUESTED** [review]
In the previous review, we requested the contributor to address certain issues. The contributor has made changes in response to the previous review, and this review pertains to those changes.

It involves modifications in one component:

1. **LocationSerachBar.vue**: The changes have been tested using the same test cases as in the initial review, and unfortunately, the application crashed again. I have attached the new logs from the browser console.
[708-LocationSerachBar.log](https://github.com/InternetHealthReport/ihr-website/files/13759202/708-LocationSerachBar.log)

We kindly ask the contributor to perform thorough testing of the proposed changes before requesting another review.

In summary, this pull request cannot be merged without first addressing the issues mentioned in this review.

**Comment by Animesh239** [comment]
hey @dpgiakatos, thanku for providing for logs.
I think the logs are indicating some error in the plotly graph.
In networkDelayChart.vue line.293
```vue
    // Add null if there is missing data
    let prevDate = new Date(trace.x.slice(-1)[0]).getTime()
    let currDate = new Date(elem.timebin).getTime()
    if (currDate > prevDate + timeResolution + 1) {
      trace.y.push(null)
      trace.x.push(elem.timbin)
    }
``` 
there is a typo error `trace.x.push(elem.timbin)`
it should be`trace.x.push(elem.timebin)`

I think this should resolve the crashing...

**Comment by dpgiakatos** [comment]
Before reviewing the changes, have you tested using the provided test case?

**Comment by Animesh239** [comment]
below are the screenshots of browser console for localhost
Screenshot (97)
Screenshot (98)


in summary, the changes are not causing any errors in the browser console.

File: src/components/search/LocationSearchBar.vue [hunk 3]
```diff
@@ -69,6 +68,15 @@ const search = (value, update) => {
         update()
         return options.value.length > MAX_RESULTS
       })
+      if (options.value.length === 0 ) {
+        // Add "No results found" option
+        options.value.push({
+          value: null,
+          name: `No results found for "${value}"`,
```
Comment (dpgiakatos) [inline src/components/search/LocationSearchBar.vue:75]: Replace `No results found for "${value}"` with `No results found`

File: src/components/search/LocationSearchBar.vue [hunk 4]
```diff
@@ -69,6 +68,15 @@ const search = (value, update) => {
...
+          label: `No results found for "${value}"`,
```
Comment (dpgiakatos) [inline src/components/search/LocationSearchBar.vue:76]: Replace `No results found for "${value}"` with `No results found`

File: src/components/search/NetworkSearchBar.vue
Comment (dpgiakatos) [inline src/components/search/NetworkSearchBar.vue:1]: This file no longer exists.

**Review by dpgiakatos — CHANGES_REQUESTED** [review]
You should rebase your repository from the dev branch as there are some conflicts, and proceed with addressing the issues highlighted in the files you changed in this PR. Also, make sure to check the code for any errors before submitting it. Thank you!
